book/src/pong-tutorial/pong-tutorial-04.md

```diff
@@ -368,7 +368,123 @@ let game_data = GameDataBuilder::default()
 You should now have a ball moving and bouncing off paddles and off the top
 and bottom of the screen. However, you will quickly notice that if the ball
 goes out of the screen on the right or the left, it never comes back
-and the game is over...
+and the game is over. You might not even see that at all, as the ball might be already
+outside of the screen when the window comes up. You might have to dramatically reduce
+`BALL_VELOCITY_X` in order to see that in action. This obviously isn't a good solution for actual game.
+To fix that problem and better see what's happening we have to spawn the ball with a slight delay.
+
+## Spwaning ball with a delay
+
+The ball now spawns and moves off screen instantly when the game starts. This might be disorienting,
+as you might be thrown into the game and loose your first point before you had the time to notice.
+We also have to give some time for the operating system and the renderer to initialise the window
+before the game starts. Usually you would have a separate state with a game menu, so this isn't an issue.
+Our pong game throws you right into the action, so we have to fix that problem.
+
+Let's delay the first time the ball spawns. This is also a good opportunity to use our game state
+struct to actually hold some data.
+
+First, let's add a new method to our state: `update`.
+Let's add that `update` method just below `on_start`:
+
+```rust,edition2018,no_run,noplaypen
+# extern crate amethyst;
+# use amethyst::prelude::*;
+# struct MyState;
+# impl SimpleState for MyState {
+fn update(&mut self, data: &mut StateData<'_, GameData<'_, '_>>) -> SimpleTrans {
+    Trans::None
+}
+# }
+```
+
+That method allows you to transition out of state,
```

Comment:
Maybe this could be reworded?
"That method allows you to transition out of state by specifying commands via the return value. Here, we do not want to change any state, so we return `Trans::None`."

Reply:
Reworded, even a bit simpler than that :)